velox/common/base/tests/StatusTest.cpp

```diff
@@ -128,5 +128,23 @@ TEST(StatusTest, macros) {
   ASSERT_TRUE(didThrow) << "VELOX_CHECK_OK did not throw";
 }
 
+Expected<int> modulo(int a, int b) {
+  if (b == 0) {
+    return folly::makeUnexpected(Status::UserError("division by zero"));
```

Comment:
the only annoying part with folly::Expected is that it doesn't let you to simply:

```
  return Status::UserError("division by zero");
```

like you can with arrow::Result. We could try to work around it, but it doesn't let you do it by default (I'm not exactly sure why)

Reply:
That's a good concern, otherwise the macros won't work with `Expect`